compiler/optimizer/abstractinterpreter/AbsLocalVarArray.cpp

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "optimizer/abstractinterpreter/AbsLocalVarArray.hpp"
+
+AbsLocalVarArray::AbsLocalVarArray(TR::Region &region) :
+      _array(region)
+   {
+   }
+
+AbsLocalVarArray::AbsLocalVarArray(AbsLocalVarArray &other, TR::Region& region) :
+      _array(region)
+   {
+   for (size_t i = 0; i < other._array.size(); i ++)
+      {
+      if (other._array.at(i))
+         {
+         _array.push_back(AbsValue::create(other._array.at(i), region));
+         }
+      else
+         {
+         _array.push_back(NULL);
+         }
+      }
+   }
+
+void AbsLocalVarArray::setToTop()
+   {
+   size_t size = _array.size();
+   for (size_t i = 0; i < size; i ++)
+      {
+      if (_array.at(i))
+         _array.at(i)->setToTop();
+      }
+   }
+   
+void AbsLocalVarArray::merge(AbsLocalVarArray &other, OMR::ValuePropagation *vp)
+   {
+   int32_t otherSize = other.size();
+   int32_t selfSize = size();
+   int32_t mergedSize = std::max(selfSize, otherSize);
+
+   for (int32_t i = 0; i < mergedSize; i++)
+      {
```

Comment:
Just a trade off in footprint since the vector will grow larger with more wasted space - likely vector would be ok here but some kind of measurement to confirm might be wise - we've had issues with vector taking a lot of space previously.